plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/IngestGeoIpPlugin.java

```diff
@@ -66,48 +69,89 @@
-            Iterator<Path> iterator = databaseFiles.iterator();
-            while (iterator.hasNext()) {
-                Path databasePath = iterator.next();
-                if (Files.isRegularFile(databasePath) && pathMatcher.matches(databasePath)) {
-                    String databaseFileName = databasePath.getFileName().toString();
-                    DatabaseReaderLazyLoader holder = new DatabaseReaderLazyLoader(
-                            databasePath,
-                            () -> {
-                                DatabaseReader.Builder builder = createDatabaseBuilder(databasePath).withCache(NoCache.getInstance());
-                                if (loadDatabaseOnHeap) {
-                                    builder.fileMode(Reader.FileMode.MEMORY);
-                                } else {
-                                    builder.fileMode(Reader.FileMode.MEMORY_MAPPED);
-                                }
-                                return builder.build();
-                            });
-                    databaseReaders.put(databaseFileName, holder);
+
+        // load any custom databases
+        if (Files.exists(geoIpConfigDirectory)) {
+            try (Stream<Path> databaseFiles = Files.list(geoIpConfigDirectory)) {
+                PathMatcher pathMatcher = geoIpConfigDirectory.getFileSystem().getPathMatcher("glob:**.mmdb");
+                // Use iterator instead of forEach otherwise IOException needs to be caught twice...
+                Iterator<Path> iterator = databaseFiles.iterator();
+                while (iterator.hasNext()) {
+                    Path databasePath = iterator.next();
+                    if (Files.isRegularFile(databasePath) && pathMatcher.matches(databasePath)) {
+                        String databaseFileName = databasePath.getFileName().toString();
+                        final DatabaseReaderLazyLoader loader = createLoader(databasePath, loadDatabaseOnHeap);
+                        databaseReaders.put(databaseFileName, loader);
+                    }
                 }
             }
         }
         return Collections.unmodifiableMap(databaseReaders);
     }
 
+    private static DatabaseReaderLazyLoader createLoader(Path databasePath, boolean loadDatabaseOnHeap) {
+        return new DatabaseReaderLazyLoader(
+                databasePath,
+                () -> {
+                    DatabaseReader.Builder builder = createDatabaseBuilder(databasePath).withCache(NoCache.getInstance());
+                    if (loadDatabaseOnHeap) {
+                        builder.fileMode(Reader.FileMode.MEMORY);
+                    } else {
+                        builder.fileMode(Reader.FileMode.MEMORY_MAPPED);
+                    }
+                    return builder.build();
+                });
+    }
+
+    private static void assertDatabaseExistence(final Path path, final boolean exists) throws IOException {
```

Comment:
Should we move this to the unit tests to allow a user the ability to remove the default database(s) if they so choose.

Reply:
Good question. I don't think we want them removing the default database files, as that means messing with the installation (poking around in the plugins directory for now, and the modules directory in the future). In fact, we have an explicit check that the default database files exist.